### PR TITLE
BuildCheckpoints: migrate to `java.time` API

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java [hunk 1]
```diff
@@ -135,17 +138,18 @@ public Integer call() throws Exception {
...
 
-        final long timeAgo = now - (86400 * days);
-        System.out.println("Checkpointing up to " + TimeUtils.dateTimeFormat(timeAgo * 1000));
+        Instant timeAgo = now.minus(days, ChronoUnit.DAYS);
+        System.out.println("Checkpointing up to " + TimeUtils.dateTimeFormat(timeAgo.toEpochMilli()));
 
         chain.addNewBestBlockListener(Threading.SAME_THREAD, block -> {
             int height = block.getHeight();
-            if (height % params.getInterval() == 0 && block.getHeader().getTimeSeconds() <= timeAgo) {
+            if (height % params.getInterval() == 0 && block.getHeader().getTimeInstant().compareTo(timeAgo) <= 0) {
```
Comment (msgilligan) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java:149]: I would use `isBefore()` or `isAfter()` here. Although using `compareTo` and `<=` looks like it's closer to the old code/easier to compare, it's actually different (I briefly thought the `0` value was replacing `timeAgo` with epoch seconds.)

**Review by msgilligan — CHANGES_REQUESTED** [review]
Minor change requested.

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java (hunk 1 above)
Comment (schildbach) [reply tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java:149]: Ok, I'll rewrite it with `isAfter()` (and the operands swapped).

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**